packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -716,7 +716,10 @@ class TextSelectionOverlay {
     final TextPosition position = renderObject.getPositionForPoint(
       Offset(details.globalPosition.dx, centerOfLineGlobal),
     );
-    _dragStartSelection ??= _selection;
+    _dragStartSelection =
+        defaultTargetPlatform == TargetPlatform.iOS || defaultTargetPlatform == TargetPlatform.macOS
+            ? _selection
+            : null;
```

Comment:
Just double checking a logical difference: after this change, on iOS/macOS, it's possible that _dragStartSelection will be overwritten when it is non-null, but previously it wouldn't have been.

Reply:
Good catch, I didn't notice that.